src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -295,10 +296,10 @@ object Task {
     *  - "myGroup_myApp.instance-b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.rails.42"
     *
     * @param instanceId Identifies the instance the task belongs to.
-    * @param containerName If set identifies the container in the pod. Defaults to [[Task.Id.Names.anonymousContainer]].
-    * @param attempt Counts how often a task has been launched on a specific reservation.
+    * @param containerName If set identifies the container. Defaults to [[Task.Id.Names.anonymousContainer]].
+    * @param attempt Counts the incarnations of the task associated with the given instance.
     */
-  case class ResidentTaskId(val instanceId: Instance.Id, val containerName: Option[String], attempt: Long) extends Id {
+  case class TaskIdWithIncarnation(val instanceId: Instance.Id, val containerName: Option[String], attempt: Long) extends Id {
```

Comment:
I'd actually take the chance an introduce a new task id type that also adds annotates the Mesos task ids so that parsing becomes simpler. So instead of

```
myGroup_myApp.instance-b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.rails.42
```

id'd like to have

```
$v1$myGroup_myApp.instance-b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.rails.42
```

or something similar. If we ever change the id we can simply change id version.

Reply:
My proposal is #6660. Look at that now it does not seem to yield much value and we can tackle the ids at a later time. It'd love to parse them with https://github.com/lihaoyi/fastparse or something similar and not regex.